### PR TITLE
Dasol user admin permissions + refactoring permissions

### PR DESCRIPTION
**Pull request by dasolyoon**
added filters to controllers so that we don't have to check permissions in every method

### CODE REVIEW

File: features/without_team_page.feature [hunk 1]
```diff
@@ -22,20 +22,21 @@ Feature: join or create a team
...
+    Then I should see "Unable to join team"
     
-  Scenario: An existing user creates a new team and gets the password
-    Given I fill in "Email" with "legueoflegends667@hotmail.com"
-    And I fill in "Password" with "password1"
-    And I press "Log In"
-    Then I should see "Create or Join a Team"
-    And I press "Create"
-    And I should see "Team Password:"
+  # Scenario: An existing user creates a new team and gets the password
```
Comment (jjlgao) [inline features/without_team_page.feature]: What's wrong with this and why is it being removed instead of fixed?

Comment (dasolyoon) [inline features/without_team_page.feature]: forgot to uncomment, thanks for catching it!

File: features/user_edit_team.feature [hunk 2]
```diff
@@ -13,14 +13,14 @@ Feature: A User edits their team
     And I log in as a user with email "eecs666@hotmail.com"
     And I am on the team_index page
  	
-  Scenario: A user removes another user from their team
-    When I press "remove_Jorge"
-    Then I should see "Removed Jorge from team."
+  # Scenario: A user removes another user from their team
```
Comment (jjlgao) [inline features/user_edit_team.feature]: oops this one should be uncommented too

**Comment by jjlgao** [comment]
lgtm unless travis fails

**Pull request merged**